subsys/usb/usb_c/usbc_pe_snk_states.c

```diff
@@ -0,0 +1,839 @@
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	LOG_INF("PE_SNK_Send_Soft_Reset");
+
+	/* Reset Protocol Layer */
+	prl_reset(dev);
+	atomic_set_bit(&pe->flags, PE_FLAGS_SEND_SOFT_RESET);
+}
+
+/**
+ * @brief PE_Send_Soft_Reset Run State
+ */
+void pe_send_soft_reset_run(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+	struct usbc_port_data *data = dev->data;
+	struct protocol_layer_rx_t *prl_rx = data->prl_rx;
+	union pd_header header;
+
+	if (prl_is_running(dev) == false) {
+		return;
+	}
+
+	if (atomic_test_and_clear_bit(&pe->flags, PE_FLAGS_SEND_SOFT_RESET)) {
+		/* Send Soft Reset message */
+		send_ctrl_msg(dev, pe->soft_reset_sop, PD_CTRL_SOFT_RESET);
+		return;
+	}
+
+	if (atomic_test_and_clear_bit(&pe->flags, PE_FLAGS_MSG_DISCARDED)) {
+		/* Inform Device Policy Manager that the message was discarded */
+		policy_notify(dev, MSG_DISCARDED);
+		pe_set_state(dev, PE_SNK_READY);
+	} else if (atomic_test_and_clear_bit(&pe->flags, PE_FLAGS_TX_COMPLETE)) {
+		/* Start SenderResponse timer */
+		usbc_timer_start(&pe->pd_t_sender_response);
+	} else if (atomic_test_and_clear_bit(&pe->flags, PE_FLAGS_MSG_RECEIVED)) {
+		header = prl_rx->emsg.header;
+
+		if (received_control_message(dev, header, PD_CTRL_ACCEPT)) {
+			pe_set_state(dev, PE_SNK_WAIT_FOR_CAPABILITIES);
+		}
+	} else if (atomic_test_bit(&pe->flags, PE_FLAGS_PROTOCOL_ERROR) ||
+		   usbc_timer_expired(&pe->pd_t_sender_response)) {
+		if (atomic_test_bit(&pe->flags, PE_FLAGS_PROTOCOL_ERROR)) {
+			atomic_clear_bit(&pe->flags, PE_FLAGS_PROTOCOL_ERROR);
+		} else {
+			policy_notify(dev, PORT_PARTNER_NOT_RESPONSIVE);
+		}
+		pe_set_state(dev, PE_SNK_HARD_RESET);
+	}
+}
+
+/**
+ * @brief PE_Send_Soft_Reset Exit State
+ */
+void pe_send_soft_reset_exit(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+
+	/* Stop Sender Response Timer */
+	usbc_timer_stop(&pe->pd_t_sender_response);
+}
+
+/**
+ * @brief PE_SNK_Soft_Reset Entry State
+ */
+void pe_soft_reset_entry(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	LOG_INF("PE_SNK_Soft_Reset");
+
+	/* Reset the Protocol Layer */
+	prl_reset(dev);
+	atomic_set_bit(&pe->flags, PE_FLAGS_SEND_SOFT_RESET);
+}
+
+/**
+ * @brief PE_SNK_Soft_Reset Run State
+ */
+void  pe_soft_reset_run(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	if (prl_is_running(dev) == false) {
+		return;
+	}
+
+	if (atomic_test_and_clear_bit(&pe->flags, PE_FLAGS_SEND_SOFT_RESET)) {
+		/* Send Accept message */
+		send_ctrl_msg(dev, PD_PACKET_SOP, PD_CTRL_ACCEPT);
+		return;
+	}
+
+	if (atomic_test_and_clear_bit(&pe->flags, PE_FLAGS_TX_COMPLETE)) {
+		pe_set_state(dev, PE_SNK_WAIT_FOR_CAPABILITIES);
+	} else if (atomic_test_and_clear_bit(&pe->flags, PE_FLAGS_PROTOCOL_ERROR)) {
```

Comment:
Out of curiosity... I've seen this bit referenced a bit, but not sure who sets it? 

Also, hard reset if our Accept was discarded?

Reply:
Good catch. I forgot to remove these PE_FLAGS_PROTOCOL_ERROR flag checks. They have been removed